cpp/src/arrow/compute/kernels/aggregate_basic.cc

```diff
@@ -532,13 +532,13 @@ struct BooleanAnyImpl : public ScalarAggregator {
     }
     if (batch[0].is_scalar()) {
       const Scalar& scalar = *batch[0].scalar;
-      this->has_nulls = !scalar.is_valid;
-      this->any = scalar.is_valid && checked_cast<const BooleanScalar&>(scalar).value;
-      this->count += scalar.is_valid;
+      this->has_nulls |= !scalar.is_valid;
```

Comment:
Is this correct? If the argument is a scalar, isn't it as if the whole batch was made from repeated values of that scalar?

Reply:
1. If previous is `has_null`, it always has_null ( that's why using `|=` )
2. `!scalar.is_valid` means has-null, so it would set by this

Reply:
I think @mapleFU is right here. `this->has_nulls` is meant to represent whether any null was encountered, so it should never change from `true` to `false`.